client.go

```diff
@@ -100,6 +100,15 @@ func (c *WorkwxApp) composeQyapiURLWithToken(path string, req interface{}, withA
 		_ = c.syncAccessToken()
 		c.tokenMu.RLock()
 	}
+	// check whether the token has expired
+	{
+		c.tokenMu.RLock()
+		expireTime := c.lastRefresh.Add((c.tokenExpiresIn - 60) * time.Second)
+		c.tokenMu.RUnlock()
+		if expireTime.Before(time.Now()) {
+			_ = c.syncAccessToken()
+		}
+	}
```

Comment:
This doesn't seem necessary. If the existing token refresher goroutine flow is used, there is already a window of 1 or 2 minutes before the current token expires, so barring extreme connectivity problems `c.accessToken` should always be ready for use.

Reply:
removed